.travis.yml

```diff
@@ -46,6 +46,13 @@ jobs:
     - name: storybook-smoke-test (node@production)
       <<: *node-prod
       script: yarn storybook-smoke-test
+    - name: yarn serve (node@production)
+      <<: *node-prod
+      script:
+        # `yarn build` takes some time so let's simulate it instead.
+        - mkdir build && touch build/index.html
+        - yarn serve &
+        - sleep 120 && pkill -f node
```

Comment:
This seems fragile. I can foresee `yarn serve` failing for other reasons, like maybe not binding to the port, etc.

Can't we just simulate the way `ts-node` is trying to compile the source?

I tried adding this file as `tsconfig-node.js`:

```js
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": ["scripts/server.tsx"]
}
```

...and running `tsc -p tsconfig-node.json` but it didn't reproduce the error. There must be some other compiler options it's adding. 

I don't want to get out of sync with how `yarn serve` actually works but I'm just wondering if there is a more direct way to check for compilation errors.

Reply:
Dont you think all reasons that fail `yarn serve` should fail the build?

Reply:
I was thinking more, like, what if the port isn't available on the CI machine. Since we don't care about running a server then, yes, this should not fail the build. I guess the port would most likely be available, though.